include/copc-lib/hierarchy/key.hpp

```diff
@@ -70,6 +78,41 @@ inline bool operator==(const VoxelKey &a, const VoxelKey &b)
 }
 inline bool operator!=(const VoxelKey &a, const VoxelKey &b) { return !(a == b); }
 
+class Box
+{
+  public:
+    Box() = default;
+    Box(const double &x_min, const double &x_max, const double &y_min, const double &y_max, const double &z_min,
+        const double &z_max)
+        : x_min(x_min), x_max(x_max), y_min(y_min), y_max(y_max), z_min(z_min), z_max(z_max){};
+    Box(const double &x_min, const double &x_max, const double &y_min, const double &y_max)
+        : x_min(x_min), x_max(x_max), y_min(y_min), y_max(y_max){};
+    Box(const VoxelKey &key, const las::LasHeader &header);
+
+    static Box ZeroBox() { return Box(0, 0, 0, 0, 0, 0); }
```

Comment:
What about ```static VoxelKey InvalidKey() { return VoxelKey(); }```?

Reply:
I actually use the default value as max so that if one isn't provided it contains everything by default. For instance for the 2D constructor the Box should contains everything on Z axis.

Reply:
I personally prefer the default constructor to be empty, like we do for the LasHeader min/max (setting them to 0). And then for the 2d constructor, x/y would be 0 and z would be the entire range. What do you think? I think it would reduce unexpected behavior more for the user if they queried a default box and it returned nothing instead of everything (i.e. signaling that they need to actually specify the extent of the box). On that note, will any of the spatial logic return true for an empty box?

Reply:
Ok this is solved, it will return true with empty box only for certain cases

if a box touches 0 it will return true for intersect and contains with the empty box. But that's normal.

Reply:
Okay, sounds good. It seems strange but I think that works with our logic.